ocaml/xapi/xapi_pci_helpers.ml

```diff
@@ -41,6 +41,32 @@ let get_driver_name address =
       debug "get_driver_name: for %s failed with %s" address (Printexc.to_string e); 
       None
 
+let address_of_dev x =
```

Comment:
We have very similar functions in pciops.ml. I can't recall who we ended up with two helper modules for PCI operations (probably refactoring due to mutual module dependencies). Perhaps it is better to merge them?

Reply:
They're also slighlty different, treating the function either as hex or decimal (which as I've been just explaind doesn't matter because the max is 7). However the usual way to format it is without a leading 0, so we should unify these PCI formatting functions (there is one in xcp-idl)

Reply:
These printing functions depend on different types and are not compatible. In this particular instance I think having a local definition is actually clearer. But I share the frustration about the many types defined around PCI devices.

Reply:
I've updated this PR:

* The previous hack around the problem is removed and will lead to an internal error if it re-appears.
* To address performance concerns, queries into sys-fs are now cached. 
* I don't understand why the problem did not surface with network SR-IOV.